src/ddmd/dstruct.d

```diff
@@ -619,9 +616,8 @@ extern (C++) class StructDeclaration : AggregateDeclaration
 
         size_t nfields = fields.dim - isNested();
         size_t offset = 0;
-        for (size_t i = 0; i < elements.dim; i++)
+        foreach(i, e; (*elements)[])
         {
-            Expression e = (*elements)[i];
```

Comment:
The segfault might be because `e` is reassigned below.  Try it without this part.